src/cpp/client.cpp

```diff
@@ -29,20 +29,30 @@
 #include <ctype.h>
 #include "client.h"
 #include "srexception.h"
+#include "logger.h"
+#include "utility.h"
 
 using namespace SmartRedis;
 
 // Constructor
-Client::Client(bool cluster)
-    : _redis_cluster(cluster ? new RedisCluster() : NULL),
-      _redis(cluster ? NULL : new Redis())
+Client::Client(bool cluster, const std::string& client_id)
 {
+    // Set up logging
+    Logger& logger = Logger::get_instance();
+    logger.configure_logging(client_id);
```

Comment:
Is there a way that we can make logger id not dependent on the last constructor called?  If we wanted to log from other objects that were just built, this seems like it would be confusing/difficult to maintain and read.

Reply:
That would be a major change. I see what you're saying though. Let's discuss this in conjunction with some of Andrew's feedback

Reply:
Marking this as resolved since it's largely the same as Andrew's comment and we have an architectural direction